sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java

```diff
@@ -443,4 +444,35 @@ private static void logOutcome(StringBuilder stringBuilder, EvaluationOutcome ou
 
         return null;
     }
+
+    @VisibleForTesting
+    UUID getTargetConfig(PodInstanceRequirement podInstanceRequirement, Collection<Protos.TaskInfo> taskInfos) {
+        if (podInstanceRequirement.getRecoveryType().equals(RecoveryType.NONE) || taskInfos.isEmpty()) {
+            return targetConfigId;
+        } else {
+            // 1. Recovery always only handles tasks with a goal state of RUNNING
+            // 2. All tasks in a pod should be launched with the same configuration
+            // Therefore it is correct to take the target configuration of one task as being
+            // representative of the whole of the pod. If tasks in the same pod with a goal
+            // state of RUNNING had different target configurations this should be rectified
+            // in any case, so it is doubly proper to choose a single target configuration as
+            // representative of the whole pod's target configuration.
+
+            Protos.TaskInfo taskInfo = taskInfos.stream().findFirst().get();
+            try {
+                return new TaskLabelReader(taskInfo).getTargetConfiguration();
+            } catch (TaskException e) {
+                logger.error(
```

Comment:
Doesn't logger automatically format the strings? I've definitely seen something like:
```
logger.error("print {} and {}", "foo", "bar");
```
So that would mean:
```java
logger.error("Falling back to current target configuration '{}'. Failed to determine target configuration for task: {}, with exception: {}", targetConfigId,  TextFormat.shortDebugString(taskInfo), e);
```
instead of the nested format.